packages/neo4j/src/account.ts

```diff
@@ -0,0 +1,96 @@
+import neo4j from "neo4j-driver"
+import type { Account } from "next-auth"
+import { v4 as uuid } from "uuid"
+
+export const linkAccount = async (
+  neo4jSession: typeof neo4j.Session,
+  data: Account
+) => {
+  const { userId, providerAccountId, provider, ...accountData } = data
+
+  // TODO: Question, account.expires is an int. here
+  // Should it be changed to js Date ?
+
+  let result
+  try {
+    result = await neo4jSession.writeTransaction((tx) =>
+      tx.run(
+        `
+      MATCH (u:User { id: $userId })
+      // Use merge here because composite of
+      // providerId + providerAccountId is unique
+      MERGE (a:Account { 
+        providerAccountId: $providerAccountId, 
+        provider: $provider 
+      }) 
+      ON CREATE SET 
+        a.id = $id
+      SET 
+        a += $accountData
+
+      MERGE (u)-[:HAS_ACCOUNT]->(a)
+
+      RETURN a, u.id AS userId
+      `,
+        {
+          id: uuid(),
```

Comment:
Would be awesome, if there was a way to generate an `id` upon creation in `neo4j`. We could get rid of the `uuid` dependency. Do you think it is possible? Most of the other adapters just create an `id` automatically, without you needing to pass it here.

Reply:
I'll look into this and speak to the neo4j community. 

My reason for not using internal ids, [neo4j docs](https://neo4j.com/docs/cypher-manual/current/clauses/match/#get-node-rel-by-id):

> Neo4j reuses its internal ids when nodes and relationships are deleted. This means that applications using, and relying on internal Neo4j ids, are brittle or at risk of making mistakes. It is therefore recommended to rather use application-generated ids.

Alternative is to use [Neo4j APOC Library](https://neo4j.com/developer/neo4j-apoc/) to move the dependency from from the app to the database. But, this may just be a different inconvenience for developers. @ndom91 and I had a chat about this and decided to go with the js package for uuid because it was 'easiest' for the user.

Reply:
@balazsorban44 do you want to have a chat about this? No-one has got back to me in the neo4j community chat. 
I feel like the `uuid` is an acceptable compromise. I don't have much more time to spend on this, so if you're not sure, then let's arrange a quick chat to work something out. Cheers.

Reply:
Thanks for checking, no problem! We can let this be for now, and handle it a future optimization opportunity.